src/gt4py/gtscript.py

```diff
@@ -111,23 +111,28 @@
 
 
 def _set_arg_dtypes(definition: Callable[..., None], dtypes: Dict[Type, Type]):
-    assert isinstance(definition, types.FunctionType)
-    annotations = getattr(definition, "__annotations__", {})
-    original_annotations = {**annotations}
-    for arg, value in annotations.items():
-        if isinstance(value, _FieldDescriptor) and isinstance(value.dtype, str):
-            if value.dtype in dtypes:
-                annotations[arg] = _FieldDescriptor(
-                    dtypes[value.dtype], value.axes, value.data_dims
+    def _parse_annotation(arg, annotation):
+        if isinstance(annotation, _FieldDescriptor) and isinstance(annotation.dtype, str):
+            if annotation.dtype in dtypes:
+                return _FieldDescriptor(
+                    dtypes[annotation.dtype], annotation.axes, annotation.data_dims
                 )
             else:
-                raise ValueError(f"Missing '{value.dtype}' dtype definition for arg '{arg}'")
-        elif isinstance(value, str):
-            if value in dtypes:
-                annotations[arg] = dtypes[value]
+                raise ValueError(f"Missing '{annotation.dtype}' dtype definition for arg '{arg}'")
+        elif isinstance(annotation, str):
+            if annotation in dtypes:
+                return dtypes[annotation]
             else:
-                raise ValueError(f"Missing '{value}' dtype definition for arg '{arg}'")
+                def_globals = getattr(definition, "__globals__", {})
+                return _parse_annotation(arg, eval(annotation, def_globals))
```

Comment:
I might be wrong, but it looks to me like this part can be fully replaced by a call to `typing.get_type_hints()` (maybe with customized `globalns` dict). Have you already tried?

Reply:
I've just tried, but it doesn't seem to work since `typing.get_hints()` seeks to convert string annotations into types, but fields are annotated using *instances* of `Field`.

Reply:
Ok, makes sense. Could you add a small comment mentioning this issue in the sources?